### PR TITLE
SPARKC-125: Allow aliases to be used with tuples

### PR DESCRIPTION
**Pull request by RussellSpitzer**

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
retest this please

**Comment by RussellSpitzer** [comment]
retest this please

**Comment by RussellSpitzer** [comment]
Yes that should be done in the other ticket, SPARKC-104. I can do that as well I thought you were interested in doing that branch though. If not I'll write it up

**Comment by pkolaczk** [comment]
+1, however this will need to be changed after SPARKC-4

**Comment by pkolaczk** [comment]
@RussellSpitzer it can't be merged and tests are failing, can you fix it?

**Comment by pkolaczk** [comment]
SPARKC-4 merged.

**Comment by RussellSpitzer** [comment]
retest this please

**Comment by RussellSpitzer** [comment]
@pkolaczk Rebased to master, changed ticket target for 1.3

**Comment by pkolaczk** [comment]
Update CHANGES.txt. Does it need documentation?

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/TupleColumnMapper.scala [hunk 1]
```diff
@@ -32,16 +31,31 @@ class TupleColumnMapper[T : TypeTag] extends ColumnMapper[T] {
...
-    
-    require(
-      getters.length == selectedColumns.length,
-      s"The number of columns ${selectedColumns.length} selected to write to ${struct.name} " +
-        s"is higher than the size of the tuple ${getters.length}")
-    
-    SimpleColumnMapForWriting(getters.toMap)
+
+    for ( colRef <- selectedColumns )
+    {
```
Comment (pkolaczk) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/TupleColumnMapper.scala]: code formatting, misplaced brace

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/TupleColumnMapper.scala [hunk 2]
```diff
@@ -32,16 +31,31 @@ class TupleColumnMapper[T : TypeTag] extends ColumnMapper[T] {
...
-        yield (name, selectedColumns(id.toInt - 1))
-    
-    require(
-      getters.length == selectedColumns.length,
-      s"The number of columns ${selectedColumns.length} selected to write to ${struct.name} " +
-        s"is higher than the size of the tuple ${getters.length}")
-    
-    SimpleColumnMapForWriting(getters.toMap)
+
+    for ( colRef <- selectedColumns )
```
Comment (pkolaczk) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/TupleColumnMapper.scala]: we're not writing spaces after ( and before )

**Comment by RussellSpitzer** [comment]
@pkolaczk Good catch, fixed it up :)

**Comment by pkolaczk** [comment]
Needs rebasing.

**Comment by RussellSpitzer** [comment]
Rebased and fixed the last few comments, I'll squash before merge

**Comment by pkolaczk** [comment]
+1

**Pull request closed without merging**